spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/SparseTensor.scala

```diff
@@ -1099,6 +1099,44 @@ object SparseTensor{
     res
   }
 
+  private def lastIndexOf[T: ClassTag](
+        array: Array[T],
+        value: T,
+        start: Int,
+        end: Int)(implicit ev: TensorNumeric[T]): Int = {
+    if (start > end) return -1
+    require(end <= array.length - 1, s"indexOf end should't exceed array size ${array.length - 1}" +
+      s", but got $end")
+    var i = start
+    while (i < end && array(i) == value) {
+      i += 1
+    }
+    if (array(i) == value) {
+      i
+    } else {
+      i - 1
+    }
+  }
+
+  private def firstIndexOf[T: ClassTag](
```

Comment:
why the name is "first..."

Reply:
find the first index who equals value. I will add some scala doc for this method.